### PR TITLE
fix: Right padding becomes negative under certain conditions

### PR DESCRIPTION
**Pull request by Robert-Louis**
GitHub Issue (If applicable): #624

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Some control had wrong layout 
image

## What is the new behavior?

image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] [Docs](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc) have been added/updated
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by carldebilly — APPROVED** [review]

**Review by iurycarlos — APPROVED** [review]

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs [hunk 1]
```diff
@@ -43,27 +43,28 @@ protected override Size MeasureOverride(Size availableSize)
...
 			Decrement(ref remainingSize, totalSpacingSize);
 		}
 
 		// 5. Calculate the size of Hug children
-		MeasureHugChildren(orientation, availableCounterSize, ref remainingSize, ref desiredCounterSize);
+		MeasureHugChildren(orientation, availableCounterSize, ref remainingSize, ref desiredCounterSize, counterPaddingSize);
 
 		// 6. Calculate the size of Filled children
-		MeasureFilledChildren(orientation, availableCounterSize, ref remainingSize, ref desiredCounterSize);
+		MeasureFilledChildren(orientation, availableCounterSize, counterPaddingSize, ref remainingSize, ref desiredCounterSize);
```
Comment (carldebilly) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs:67]: Why the parameter ordering is different from those 2 methods? It's purely cosmetic, but that's nagging me ;-)

**Review by carldebilly — APPROVED** [review]

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs (hunk 1 above)
Comment (Robert-Louis) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs:67]: fair enought I will fixe it ^^

**Comment by agneszitte** [comment]
@mergifyio backport legacy/3x

**Pull request merged**